### PR TITLE
Initial fuzzing setup

### PR DESCRIPTION
**Pull request by poljar**
This PR introduces an AFL based fuzzing setup for vodozemac.

Not many things are yet fuzzed, a comprehensive list of codepaths we'll want to fuzz:
- [x] Olm message decryption
- [x] Olm pre-key message decoding
- [x] Olm non-pre-key message decoding
- [x] Account [unpickling](https://github.com/matrix-org/vodozemac/blob/main/src/olm/account/mod.rs#L459)
- [ ] Session unpickling
- [ ] Olm [unpickling from a libolm pickle](https://github.com/matrix-org/vodozemac/blob/main/src/olm/account/mod.rs#L330)
- [x] Megolm [decryption](https://github.com/matrix-org/vodozemac/blob/main/src/megolm/inbound_group_session.rs#L185)
- [x] Megolm inbound session [creation](https://github.com/matrix-org/vodozemac/blob/main/src/megolm/inbound_group_session.rs#L106)
- [x] Megolm session [import](https://github.com/matrix-org/vodozemac/blob/main/src/megolm/inbound_group_session.rs#L110)
- [ ] Megolm [unpickling](https://github.com/matrix-org/vodozemac/blob/main/src/megolm/inbound_group_session.rs#L338)
- [ ] Megolm unpickling from a [libolm pickle](https://github.com/matrix-org/vodozemac/blob/main/src/megolm/inbound_group_session.rs#L252)
- [x] Megolm message [decoding](https://github.com/matrix-org/vodozemac/blob/main/src/megolm/message.rs#L33)

### CODE REVIEW

**Comment by poljar** [comment]
Let's merge this and I'll transfer the rest of the remaining tasks into an issue.

File: afl/olm-account-unpickling/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    fuzz!(|data: &[u8]| {
+        let _ = vodozemac::olm::Account::from_decrypted_libolm_pickle(&data);
+    });
+}
+
+#[cfg(feature = "cli")]
+use clap::Parser;
+
+#[cfg(feature = "cli")]
+/// Search for a pattern in a file and display the lines that contain it.
```
Comment (dkasak) [inline afl/olm-account-unpickling/src/main.rs:15]:
```suggestion
```

Stray comment from example?

File: afl/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+Build the harness using the cargo afl command:
+
+```bash
+$ cargo afl build
+```
+
+Start fuzzing using the cargo afl command:
+
+```bash
+$ cargo afl fuzz -i in -o out target/debug/vodozemac-afl
```
Comment (dkasak) [inline afl/README.md]:
```suggestion
$ cargo afl fuzz -i in -o out target/debug/olm-message-decoding
```

File: afl/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+$ cd afl/olm-message-decoding
+```
+
+Build the harness using the cargo afl command:
+
+```bash
+$ cargo afl build
+```
+
+Start fuzzing using the cargo afl command:
```
Comment (dkasak) [inline afl/README.md]:
```suggestion
Start fuzzing using the `cargo afl` command, for example:
```

File: afl/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# Fuzzing
+
+To start fuzzing using one of the provided harnesses enter the subdirectory of
+the harness:
+
+```bash
+$ cd afl/olm-message-decoding
+```
+
+Build the harness using the cargo afl command:
```
Comment (dkasak) [inline afl/README.md]:
```suggestion
Build the harness using the `cargo afl` command:
```

File: afl/megolm-decoding/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "vodozemac-afl"
```
Comment (dkasak) [inline afl/megolm-decoding/Cargo.toml]:
```suggestion
name = "megolm-decoding"
```

File: afl/megolm-decryption/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "vodozemac-afl"
```
Comment (dkasak) [inline afl/megolm-decryption/Cargo.toml]:
```suggestion
name = "megolm-decryption"
```

File: afl/megolm-session-creation/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "vodozemac-afl"
```
Comment (dkasak) [inline afl/megolm-session-creation/Cargo.toml]:
```suggestion
name = "megolm-session-creation"
```

File: afl/megolm-session-import/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "vodozemac-afl"
```
Comment (dkasak) [inline afl/megolm-session-import/Cargo.toml]:
```suggestion
name = "megolm-session-import"
```

**Review by dkasak — APPROVED** [review]
Looks good!

Added some tweaks myself instead of making suggestions since Github review UI is broken by the large number of files. I think there are just two suggestions left to apply, in [afl/olm-account-unpickling/src/main.rs](https://github.com/matrix-org/vodozemac/pull/12/files#r1035912314).

File: afl/olm-account-unpickling/src/main.rs (hunk 1 above)
Comment (poljar) [reply afl/olm-account-unpickling/src/main.rs:15]: Yep.

**Pull request merged**